// @uifabric/fluent-theme/lib-commonjs/fluent/styles/Checkbox.styles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckboxStyles = function (props) {
    var disabled = props.disabled, checked = props.checked, theme = props.theme;
    var semanticColors = theme.semanticColors, palette = theme.palette, effects = theme.effects;
    return {
        checkbox: [
            {
                borderRadius: effects.roundedCorner2,
                borderColor: palette.neutralPrimary,
            },
            !disabled &&
                checked && {
                // using semanticColor because of setting a new color above
                borderColor: semanticColors.inputBackgroundChecked,
            },
            disabled && {
                borderColor: palette.neutralTertiaryAlt,
            },
            checked &&
                disabled && {
                background: palette.neutralTertiaryAlt,
                borderColor: palette.neutralTertiaryAlt,
            },
        ],
        checkmark: {
            // using semanticColor to override original behavior on checked/disabled
            color: semanticColors.inputForegroundChecked,
        },
        root: [
            !disabled && [
                !checked && {
                    selectors: {
                        ':hover .ms-Checkbox-text': { color: palette.neutralDark },
                        ':hover .ms-Checkbox-checkmark': { color: palette.neutralSecondary },
                    },
                },
                checked && {
                    selectors: {
                        ':hover .ms-Checkbox-checkbox': {
                            background: palette.themeDark,
                            borderColor: palette.themeDark,
                        },
                        ':focus .ms-Checkbox-checkbox': {
                            background: palette.themeDark,
                            borderColor: palette.themeDark,
                        },
                    },
                },
            ],
        ],
    };
};
